app/src/main/java/com/humara/nagar/ui/residents/ResidentsFragment.kt

```diff
@@ -1,18 +1,35 @@
 package com.humara.nagar.ui.residents
 
 import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
-import androidx.lifecycle.ViewModelProvider
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.chip.Chip
+import com.humara.nagar.Logger
+import com.humara.nagar.R
+import com.humara.nagar.adapter.AllResidentsAdapter
 import com.humara.nagar.analytics.AnalyticsData
 import com.humara.nagar.base.BaseFragment
+import com.humara.nagar.base.ViewModelFactory
 import com.humara.nagar.databinding.FragmentResidentsBinding
+import com.humara.nagar.ui.residents.model.FiltersResponse
+import com.humara.nagar.ui.residents.model.Residents
+import com.humara.nagar.utils.Utils
+
 
 class ResidentsFragment : BaseFragment() {
 
     private var _binding: FragmentResidentsBinding? = null
+    private val residentsViewModel by viewModels<ResidentsViewModel> {
+        ViewModelFactory()
+    }
+    private lateinit var recyclerView: RecyclerView
```

Comment:
no need of this local variable. You can use residentRCV.apply{}

Reply:
Resolved.